lib/src/command/add.dart

```diff
@@ -349,37 +432,137 @@ class AddCommand extends PubCommand {
         GitDescription(
           url: parsed.toString(),
           containingDir: p.current,
-          ref: gitRef,
-          path: gitPath,
+          ref: argResults.gitRef,
+          path: argResults.gitPath,
         ),
       );
     } else if (path != null) {
       ref = PackageRef(
           packageName, PathDescription(p.absolute(path), p.isRelative(path)));
-    } else if (sdk != null) {
-      ref = cache.sdk.parseRef(packageName, sdk);
+    } else if (argResults.sdk != null) {
+      ref = cache.sdk.parseRef(packageName, argResults.sdk);
     } else {
       ref = PackageRef(
         packageName,
         HostedDescription(
           packageName,
-          hostUrl ?? cache.hosted.defaultUrl,
+          argResults.hostUrl ?? cache.hosted.defaultUrl,
         ),
       );
     }
-    return _ParseResult(ref, constraint);
+    return _PartialParseResult(ref, constraint);
+  }
+
+  /// Parse [package] to return the corresponding [_ParseResult].
+  ///
+  /// [package] must be written in the format
+  /// `dev:<package-name>[:descriptor>]`, where quotations should be used if
```

Comment:
```suggestion
  /// `<package-name>[:descriptor>]`, where quotations should be used if
```

You've already handled this.